invenio_files_rest/tasks.py

```diff
@@ -195,20 +195,22 @@ def migrate_file(src_id, location_name, post_fixity_check=False):
 
 
 @shared_task(ignore_result=True)
-def remove_file_data(file_id, silent=True):
+def remove_file_data(file_id, silent=True, force=False):
     """Remove file instance and associated data.
 
     :param file_id: The :class:`invenio_files_rest.models.FileInstance` ID.
     :param silent: It stops propagation of a possible raised IntegrityError
         exception. (Default: ``True``)
+    :param force: Whether to delete the file even if the file instance is not
+        marked as writable.
     :raises sqlalchemy.exc.IntegrityError: Raised if the database removal goes
         wrong and silent is set to ``False``.
     """
     try:
         # First remove FileInstance from database and commit transaction to
         # ensure integrity constraints are checked and enforced.
         f = FileInstance.get(file_id)
-        if not f.writable:
+        if not f.writable and not force:
```

Comment:
InvenioRDM marks files as not writable, e.g. here's two files that i've uploaded for a draft, which i deleted afterwards:

```ipython
In [1]: from invenio_files_rest.models import ObjectVersion, FileInstance, Bucket

In [2]: orphans = FileInstance.query.filter(~FileInstance.objects.any()).all()

In [3]: orphans
Out[3]:
[<FileInstance 4d22743a-282b-4f91-956e-f8a3fc632403>,
 <FileInstance 3f3e1e10-918b-4070-9a2f-e61de0205cc1>]

In [4]: orphans[0].writable
Out[4]: False

In [5]: orphans[1].writable
Out[5]: False
```

Reply:
~~When does InvenioRDM mark files as not writable? Always? Or only when the draft is deleted or marked as deleted?~~
I re-read your explanations, now I have understood. Do you know if there are other cases?

Reply:
From the top of my head I can't remember any times when InvenioRDM would've set the files to `writable=True`... Gotta check through the codepaths again, to see if it's possible to change that behaviour.

Though likely already running instances would have (at least some, if not all) files "non-writable" already, like in our case.